utils/wasm-builder/src/wasm_project.rs

```diff
@@ -477,7 +494,32 @@ fn compact_wasm_file(
 	let bloaty_file = project.join(bloaty_file_name);
 	fs::copy(wasm_file, &bloaty_file).expect("Copying the bloaty file to the project dir.");
 
-	(wasm_compact_file, WasmBinaryBloaty(bloaty_file))
+	(
+		wasm_compact_file,
+		wasm_compact_compressed_file,
+		WasmBinaryBloaty(bloaty_file),
+	)
+}
+
+fn compress_wasm(
+	wasm_binary_path: &Path,
+	compressed_binary_out_path: &Path,
+) {
+	const MAX_UNCOMPRESSED_SIZE: usize = 100 * 1024 * 1024;
+
+	let data = fs::read(wasm_binary_path).expect("Failed to read WASM binary");
+	if let Some(compressed) = sp_maybe_compressed_blob::compress(
+		&data,
+		MAX_UNCOMPRESSED_SIZE,
+	) {
+		fs::write(compressed_binary_out_path, &compressed[..])
+			.expect("Failed to write WASM binary");
+	} else {
+		println!(
+			"Writing uncompressed wasm. Exceeded maximum size {}",
```

Comment:
Where do we write this?

If the compression fails, no file will exist, but you continue and assume the file exists.

Reply:
Where would be the best place to log an error?